Lib/pstats.py

```diff
@@ -298,14 +304,35 @@ def strip_dirs(self):
         old_top = self.top_level
         self.top_level = new_top = set()
         for func in old_top:
-            new_top.add(func_strip_path(func))
+            new_top.add(strip_function(func))
 
         self.max_name_len = max_name_len
 
         self.fcn_list = None
         self.all_callees = None
         return self
 
+    def strip_non_unique_dirs(self):
+        full_paths = set()
+
+        for (full_path, _, _), (_, _, _, _, callers) in self.stats.items():
+            full_paths.add(full_path)
+            for (full_path_caller, _, _), _ in callers.items():
+                full_paths.add(full_path_caller)
+
+        minimal_path_by_full_path = _build_minimal_path_by_full_path(full_paths)
+
+
+        for k, v in minimal_path_by_full_path.items():
+            print(f'{k} -> {v}')
+        asd()
+
+        def strip_function(fullpath_linenum_funcname):
+            fullpath, linenum, funcname = fullpath_linenum_funcname
+            return minimal_path_by_full_path[fullpath], linenum, funcname
+
+        return self._strip_directory_data(strip_function)
+
```

Comment:
here are several things that confuse me about this:

* Why it prints? It should not print anything.
* What is asd() in line 328? (Is not defined anywhere)

Reply:
Gah, so sorry. This was some debugging code that I managed to reintroduce with the rebase. I'm so sorry.